### PR TITLE
Updated doc for `admin_bar_menu`

### PR DESCRIPTION
**Pull request by narenin**
Trac ticket: https://core.trac.wordpress.org/ticket/61738

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/admin-bar.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ function wp_admin_bar_render() {
 	/**
 	 * Loads all necessary admin bar items.
 	 *
-	 * This is the hook used to add, remove, or manipulate admin bar items.
+	 * This is the hook used to add, remove, or manipulate admin bar items. The recommended hook instead of this is wp_before_admin_bar_render.
```
Comment (sabernhardt) [inline src/wp-includes/admin-bar.php]:
```suggestion
	 * This hook can add, remove, or manipulate admin bar items. The priority
	 * determines the placement for new items, and changes to existing items
	 * would require a high priority. To remove or manipulate existing nodes
	 * without a specific priority, use `wp_before_admin_bar_render`.
```

This should be the preferred hook for adding items, and it still can modify or remove items if done with a proper priority level.

**Review by sabernhardt — CHANGES_REQUESTED** [review]
Thanks for the PR!

1. I suggested an edit for the `admin_bar_menu` hook description.
2. For `wp_admin_bar_render()`, I would suggest simply removing the incorrect sentence instead of replacing it. I also added a comma before "which" and adjusted the line length after removing the sentence:
```
 * It includes the {@see 'admin_bar_menu'} action, which should be used to hook in and
 * add new menus to the admin bar. This also gives you access to the `$post` global,
 * among others.
```
3. The `wp_before_admin_bar_render` hook probably should explain more in the docblock, but the [More Information](https://developer.wordpress.org/reference/hooks/wp_before_admin_bar_render/#more-information) section might explain it better than I can.

File: src/wp-includes/admin-bar.php (hunk 1 above)
Comment (narenin) [reply src/wp-includes/admin-bar.php]: @sabernhardt Thanks for the detailed feedback. I have implemented the suggestions, please review the same.

Comment (hellofromtonya) [reply src/wp-includes/admin-bar.php]: Checked and the requested changes are now implemented in the PR ✅

Comment (sabernhardt) [reply src/wp-includes/admin-bar.php]: Thanks for updating the `admin_bar_menu` docblock.

Could you also remove the sentence about the "most optimal point" in `wp_admin_bar_render()` and limit the line length to 80 characters?
https://github.com/WordPress/wordpress-develop/blob/b50a842d4889f2e635b161f4c2d14d078651bf28/src/wp-includes/admin-bar.php#L64-L67

```
 * It includes the {@see 'admin_bar_menu'} action, which should be used to hook in and
 * add new menus to the admin bar. This also gives you access to the `$post` global,
 * among others.
```

Comment (hellofromtonya) [reply src/wp-includes/admin-bar.php]: Done in commit https://github.com/WordPress/wordpress-develop/pull/7222/commits/cb0c76490cb7b3b7af11af7d9d06459d83488b90 ✅

**Comment by hellofromtonya** [comment]
@sabernhardt @joedolson can you do another review please? Is it ready for commit?

**Review by sabernhardt — APPROVED** [review]
Looks good

**Review by joedolson — APPROVED** [review]

**Comment by hellofromtonya** [comment]
Thank you everyone for your contributions 🌟 This PR was committed via https://core.trac.wordpress.org/changeset/58978.

**Pull request closed without merging**